internal/configmaps/configmaps.go

```diff
@@ -0,0 +1,109 @@
+package configmaps
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/jakub-dzon/k4e-operator/api/v1alpha1"
+	"github.com/jakub-dzon/k4e-operator/internal/k8sclient"
+	"github.com/jakub-dzon/k4e-operator/internal/utils"
+	"github.com/jakub-dzon/k4e-operator/models"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate mockgen -package=configmaps -destination=mock_configmaps.go . ConfigMap
+type ConfigMap interface {
+	Fetch(ctx context.Context, deployments []v1alpha1.EdgeDeployment, namespace string) (models.ConfigmapList, error)
+}
+
+type configMap struct {
+	client k8sclient.K8sClient
+}
+
+func NewConfigMap(client k8sclient.K8sClient) ConfigMap {
+	return &configMap{client: client}
+}
+
+func (cm *configMap) Fetch(ctx context.Context, deployments []v1alpha1.EdgeDeployment, namespace string) (models.ConfigmapList, error) {
+	list := models.ConfigmapList{}
+
+	// create map of configmap names and keys
+	cmMap := utils.MapType{}
+	for _, deployment := range deployments {
+		podSpec := deployment.Spec.Pod.Spec
+		allContainers := append(podSpec.InitContainers, podSpec.Containers...)
+		for _, container := range allContainers {
+			extractConfigMapsFromEnv(&container, cmMap)
+
+			// Extract info also from volumes:
+			utils.ExtractInfoFromVolume(podSpec.Volumes, cmMap, func(i interface{}) (bool, *bool, string) {
+				volume := i.(corev1.Volume)
+				if volume.ConfigMap != nil {
+					return true, volume.ConfigMap.Optional, volume.ConfigMap.Name
+				}
+				return false, nil, ""
+			})
+		}
+	}
+
+	// read configmaps and add to configmaps list
+	for name, keys := range cmMap {
+		configmapObj, err := cm.readAndValidateConfigMap(ctx, name, namespace, keys)
+		if err != nil {
+			return nil, err
+		}
+		if configmapObj == nil {
+			continue
+		}
+		configmapObj.ObjectMeta = metav1.ObjectMeta{Name: name}
```

Comment:
the namespace should be added, no? I mean, you return the list, there is no way to get the same name?

Reply:
We don't need the namespace in the podman.

Reply:
in this case how do we handle name collisions? Once approval process will be implemented we will support many namespaces.

Reply:
The only way we could solve it is to change the name with the prefix with namespace.
So the ConfigMap in podman would be for example:
```yaml
kind: ConfigMap
metadata:
  name: configmapnamespace_configmapname
```

But we would need to in that case modify the podspec definition of the pod on device side, because the `configMapRef` would not be the same as k8s spec. In kubernetes world user can use only configmaps from same namespace as pod is running. That's why there is no specification of configMapRef::namespace. So maybe the concept of using only configmaps which are in same namespace as device is meaningfull here.

Reply:
AFAIR in case of Pods ConfigMap has to be in the same namespace as the Pod; I don't think we need to diverge from that and that implies that collisions would be impossible.

Reply:
That should be the same namespace as of the EdgeDeployment.